datahub-web-react/src/CustomThemeProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from 'react';
+import { ThemeProvider } from 'styled-components';
+import { Theme } from './conf/theme/types';
+import defaultThemeConfig from './conf/theme/theme_light.config.json';
+import { CustomThemeContext } from './customThemeContext';
+
+const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
+    const [currentTheme, setTheme] = useState<Theme>(defaultThemeConfig);
+
+    useEffect(() => {
+        if (import.meta.env.DEV) {
+            import(/* @vite-ignore */ `./conf/theme/${import.meta.env.REACT_APP_THEME_CONFIG}`).then((theme) => {
+                setTheme(theme);
+            });
+        } else {
+            // Send a request to the server to get the theme config.
+            fetch(`/assets/conf/theme/${import.meta.env.REACT_APP_THEME_CONFIG}`)
```

Comment:
is there a need to serialize and cache the theme?

Reply:
not sure - this is the new semantic fro vite that harshal had introduced but i don't think it's necessary..